api/v1beta1/grafanafolder_types.go

```diff
@@ -199,3 +192,18 @@ func (in *GrafanaFolder) ResyncPeriodHasElapsed() bool {
 	deadline := in.Status.LastResync.Add(in.GetResyncPeriod())
 	return time.Now().After(deadline)
 }
+
+func (in *GrafanaFolder) MatchLabels() *metav1.LabelSelector {
+	return in.Spec.InstanceSelector
+}
+
+func (in *GrafanaFolder) MatchNamespace() string {
+	return in.ObjectMeta.Namespace
+}
+
+func (in *GrafanaFolder) AllowCrossNamespace() bool {
+	if in.Spec.AllowCrossNamespaceImport != nil {
```

Comment:
I attempted this when I was implementing the CommonSpec to begin with.
In the end I reverted the changes due to a bug that now re-appeared, which may block this from being merged.

When applying the testdata example resources, a permanent update/change to GrafanaAlertRuleGroup specifically:
![image](https://github.com/user-attachments/assets/5bbe4656-3fe4-4382-9c8a-ed346bbb6bb4)

The operator will add `allowCrossNamespaceImport: false` to the spec if omitted from the applied manifest:
```diff
>   allowCrossNamespaceImport: false
```
`allowCrossNamespaceImport` is missing in: https://github.com/grafana/grafana-operator/blob/master/tests/example-resources.yaml#L52-L57

Specifically the [deferred Update](https://github.com/grafana/grafana-operator/blob/master/controllers/grafanaalertrulegroup_controller.go#L104) function is the one adding the key.
commenting out the `Update`, neither the `finalizer` nor the key is added.

Comparing other resources with similar reconciliation loops like ContactPoint and NotificationPolicy, I don't see the same behaviour and I'm stumped.

From what I know, this is not a grim bug, FluxCD Kustomization only emits events on added or removed files.
Helm will likely persist it thanks to the 3-way-merge, and it's the default value in the operator/CRDs.
It's mainly annoying when applying resources directly.

PS. Applying this change made the PR balloon from 6 to 26 changed files.

Reply:
If this could cause unintended side effects, I'd put it in a separate PR as you recommended. Just makes it easier to discuss it and trace back issues (if there are any). Thanks for exploring this!

Reply:
Yeah, I agree. Thanks for trying it out!

Reply:
I've created a Draft PR at #1776 containing the full history of this PR as well, will rebase it once this is merged.